api/siad/api.go

```diff
@@ -37,3 +37,8 @@ type SyncerPeerResponse struct {
 type SyncerConnectRequest struct {
 	NetAddress string `json:"netAddress"`
 }
+
+// A MinerAddressRequest is for setting the miner's address.
+type MinerAddressRequest struct {
+	Address types.Address `json:"address"`
+}
```

Comment:
not convinced this functionality is necessary. If you're a "real" miner, then yeah, you might want to change your address, but if you're running a CPU miner for testing, setting the address once at startup should be sufficient.